cloud/cloud.go

```diff
@@ -77,33 +74,49 @@ func promptUsername() string {
 	return username
 }
 
-type authResponse struct {
-	VMHostname string `json:"vm_host"`
+type vm struct {
+	JumpHost     string `json:"jump_host"`
+	JumpHostPort int    `json:"jump_host_port"`
+	VMHost       string `json:"vm_host"`
+	VMHostPort   int    `json:"vm_host_port"`
+	VMRegion     string `json:"vm_region"`
+	VMPublicKey  string `json:"vm_public_key"`
+	VMPrivateKey string `json:"vm_private_key"`
 }
 
 func getVirtualMachine(username string) string {
+	client := openssh.Client{
+		Username: username,
+		Hostname: "gateway.devbox.sh",
+	}
+
 	// When developing we can use this env variable to point
 	// to a different gateway
-	hostname := os.Getenv("DEVBOX_GATEWAY")
-	if hostname == "" {
-		hostname = "gateway.devbox.sh"
-	}
-	client := sshclient.Client{
-		Username: username,
-		Hostname: hostname,
+	if envGateway := os.Getenv("DEVBOX_GATEWAY"); envGateway != "" {
+		client.Hostname = envGateway
 	}
 	bytes, err := client.Exec("auth")
 	if err != nil {
-		log.Fatal(err)
+		log.Println(err)
+		var exitErr *exec.ExitError
+		if errors.As(err, &exitErr) {
+			log.Println("ssh gateway.devbox.sh stderr:\n", string(exitErr.Stderr))
```

Comment:
Since we can change the hostname of the gateway via an env variable, do you want to pass in `client.Hostname` instead of hardcoding it?
```suggestion
			log.Println("ssh %s stderr:\n", client.Hostname, string(exitErr.Stderr))
```

Reply:
Good idea. I parameterized it in the debug log for stdout too.